server/models/User.js

```diff
@@ -0,0 +1,23 @@
+const mongoose = require("mongoose");
+const { Schema } = mongoose;
+
+const UserSchema = new Schema({
+    name: {
+        type: String,
+        required: true
+    },
+    email: {
+        type: String,
+        required: true
+    },
+    password: {
+        type: String,
+        require: true
+    },
+    date: {
+        type: Date,
+        default: Date.now
+    }
```

Comment:
do you want to store projects as an array on the user with references to project id keys

Reply:
The decision will probably depend on how many projects you imagine per user - I'd probably think there are not many

Reply:
I don't think they would have many projects? I'm open to storing them as an array of projectIds on the user. What would be the alternative be?

Reply:
What do you think about adding unique (and other) validators on username and email?

Reply:
Yeah we should definitely validators so that there aren't duplicate or invalid usernames and emails